### ISSUE TITLE
TypeScript: No overload matches this call 

### ISSUE THREAD

**Issue by swalahamani** [satya164/react-native-tab-view#1161]
### Current behaviour

Getting the below error while using TypeScript:

```TypeScript
(alias) class TabView<T extends Route>
import TabView
No overload matches this call.
  Overload 1 of 2, '(props: Props<Route> | Readonly<Props<Route>>): TabView<Route>', gave the following error.
    Type '{ navigationState: { index: number; routes: Route[]; }; onIndexChange: (tabViewIndex: number) => void; renderScene: ({ route, jumpTo, position }: SceneRendererProps & { route: any; }) => Element; renderTabBar: (tabBarProps: SceneRendererProps & { ...; }) => Element; }' is missing the following properties from type 'Readonly<Props<Route>>': renderLazyPlaceholder, tabBarPosition, lazy, lazyPreloadDistance
  Overload 2 of 2, '(props: Props<Route>, context: any): TabView<Route>', gave the following error.
    Type '{ navigationState: { index: number; routes: Route[]; }; onIndexChange: (tabViewIndex: number) => void; renderScene: ({ route, jumpTo, position }: SceneRendererProps & { route: any; }) => Element; renderTabBar: (tabBarProps: SceneRendererProps & { ...; }) => Element; }' is missing the following properties from type 'Readonly<Props<Route>>': renderLazyPlaceholder, tabBarPosition, lazy, lazyPreloadDistancets(2769)
```

### Expected behaviour

This error is not supposed to be shown.

### Code sample

For Syntax Highlighting check this [link](https://help.github.com/en/articles/creating-and-highlighting-code-blocks)
```

	renderScenes = SceneMap({
		"first": FirstView,
		"second": SecondView,
		"third": ThirdView,
		"fourth": FourthView,
	});

	render() {
		// typeof tabViewIndex is number & tabViewRoutes is Array<Route>
		const {tabViewIndex, tabViewRoutes} = this.state;

		return (
			<View style={{...}} >
				.
				.
				.
				<TabView
					navigationState={{index: tabViewIndex, routes: tabViewRoutes}}

					// type of this.setTabViewIndex is (tabViewIndex: number) => void
					onIndexChange={this.setTabViewIndex}

					renderScene={this.renderScenes}

					// type of this.renderTabBar is (tabBarProps: SceneRendererProps & {
					//		navigationState: NavigationState<Route>;
					//	}) => JSX.Element
					renderTabBar={this.renderTabBar}
				/>
			</View>
		);
	}
}
```

### Screenshots (if applicable)

![image](https://user-images.githubusercontent.com/21193037/113175784-6a71db00-9269-11eb-9cd2-af2e5839ccf9.png)

### What have you tried

Cross-checked type of the props used.

### Your Environment

| software                     | version
| ---------------------------- | -------
| ios or android               | NA
| expo                         | NA
| react-native                 | 0.63.2
| react-native-tab-view        | ^3.0.0
| react-native-pager-view      | ^5.1.2
| node                         | 14.15.4
| npm or yarn                  | yarn - 1.22.10

**Comment by swalahamani** [comment]
> Couldn't find version numbers for the following packages in the issue:
> 
> * `expo`
> 
> Can you update the issue to include version numbers for those packages? The version numbers must match the format 1.2.3.
> 
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `react-native` (found: `0.63.2`, latest: `0.64.0`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

I'm not using expo.

And also, I don't think it is related to react-naitve version as it is a pure TypeScript type error.

**Comment by quachdev** [comment]
This issue looks to be fixed in [v3.0.1](https://github.com/satya164/react-native-tab-view/compare/v3.0.0...v3.0.1#diff-ea74a4a5384ccd3c5ddf2a810b95844ff6b996d576e3b6118828e7250a256a40)

Thank you!
